mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/migration/V1ToV2MigrationThread.java

```diff
@@ -0,0 +1,118 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail.migration;
+
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.james.mailbox.cassandra.ids.CassandraMessageId;
+import org.apache.james.mailbox.cassandra.mail.AttachmentLoader;
+import org.apache.james.mailbox.cassandra.mail.CassandraMessageDAO;
+import org.apache.james.mailbox.cassandra.mail.CassandraMessageDAOV2;
+import org.apache.james.mailbox.cassandra.mail.MessageAttachmentRepresentation;
+import org.apache.james.mailbox.cassandra.mail.MessageWithoutAttachment;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.store.mail.MessageMapper;
+import org.apache.james.mailbox.store.mail.model.impl.SimpleMailboxMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.EvictingQueue;
+
+public class V1ToV2MigrationThread implements Runnable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(V1ToV2MigrationThread.class);
+    public static final int POLL_INTERVAL_IN_MS = 10;
+
+    private final EvictingQueue<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> messagesToBeMigrated;
+    private final CassandraMessageDAO messageDAOV1;
+    private final CassandraMessageDAOV2 messageDAOV2;
+    private final AttachmentLoader attachmentLoader;
+
+    public V1ToV2MigrationThread(EvictingQueue<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> messagesToBeMigrated,
+                                 CassandraMessageDAO messageDAOV1, CassandraMessageDAOV2 messageDAOV2, AttachmentLoader attachmentLoader) {
+        this.messagesToBeMigrated = messagesToBeMigrated;
+        this.messageDAOV1 = messageDAOV1;
+        this.messageDAOV2 = messageDAOV2;
+        this.attachmentLoader = attachmentLoader;
+    }
+
+    @Override
+    public void run() {
+        while (true) {
+            try {
+                Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>> message = dequeue();
+                performV1ToV2Migration(message).join();
+            } catch (Exception e) {
+                LOGGER.error("Error occured in migration thread", e);
+            }
+        }
+    }
+
+    private Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>> dequeue() {
+        while (true) {
+            Optional<Pair<MessageWithoutAttachment, Stream<MessageAttachmentRepresentation>>> poll = poll();
```

Comment:
I propose you a commit to replace it by https://crunchify.com/arrayblockingqueue-vs-google-guava-non-blocking-evictingqueue-example/.
This has the effect of evicting new message instead of old one but I guess we don't really care about that